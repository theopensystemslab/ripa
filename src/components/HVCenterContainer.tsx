import Box from "@material-ui/core/Box";
import Container from "@material-ui/core/Container";
import { makeStyles } from "@material-ui/core/styles";
import * as React from "react";

import { scrollIn } from "../lib/scrollIn";

const useStyles = makeStyles(theme => ({
  root: {
    minHeight: "100vh",
    "&:first-child": {
      minHeight: "calc(100vh - 168px)" // 100vh - header
    },
    "&:only-child": {
      minHeight: "calc(100vh - 220px)" // 100vh - header + footer
    },
    "&:last-child:not(:first-child)": {
      minHeight: "calc(100vh - 116px)" // 100vh - small header + footer
    }
  },
  py: {
    paddingTop: theme.spacing(3),
    paddingBottom: theme.spacing(3),
    [theme.breakpoints.up("lg")]: {}
  }
}));

const HVCenterContainer = ({
  children,
  light = false,
  verticalCenter = false,
  disableScroll = false
}) => {
  const ref = React.useRef(null);
  React.useLayoutEffect(() => {
    if (!disableScroll) {
      scrollIn(ref.current, {
        block: verticalCenter ? "center" : "start"
      });
    }
  });
  const classes = useStyles();
  return (
    <Box
      className={classes.root}
      color={light ? "#000" : "#fff"}
      bgcolor={light ? "#fff" : "primary.main"}
      style={{
        alignItems: verticalCenter && "center",
        display: verticalCenter && "flex"
      }}
    >
      <Container
        className={verticalCenter ? "vertical-center" : classes.py}
        maxWidth="md"
        ref={ref}
      >
        {children}
      </Container>
    </Box>
  );
};

export default HVCenterContainer;
