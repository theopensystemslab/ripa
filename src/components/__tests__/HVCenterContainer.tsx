import { cleanup, render } from "@testing-library/react";
import * as React from "react";

import HVCenterContainer from "../HVCenterContainer";

afterEach(cleanup);

describe("HVCenterContainer Component", () => {
  it("should render snapshot", () => {
    const children = "This should be rendered inside";
    const { asFragment } = render(
      <HVCenterContainer verticalCenter={true} light={true}>
        {children}
      </HVCenterContainer>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
