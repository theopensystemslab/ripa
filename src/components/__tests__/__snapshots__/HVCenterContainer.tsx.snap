// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HVCenterContainer Component should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-3 makeStyles-root-1"
    style="align-items: center; display: flex;"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd"
    >
      This should be rendered inside
    </div>
  </div>
</DocumentFragment>
`;
